### PR TITLE
Fix Broken Logic for Loading Profile Fields

### PR DESCRIPTION
**Pull request by colinschoen**
I noticed these issues when investigating issue #4451. This doesn't fix that issue, but addresses other issues in PR #4383. 

Specifically this adds the `"options":["None","Male","Female"]` key to the serialized JSON for the `displayFields` setting. 

It also fixes some logic where the `currentKey` variable was being improperly set for custom fields with multiple options.

@MissAllSunday I echo the concerns of @albertlast after looking into this more... are you sure that we don't need any changes in the upgrade script? If you can please take a look to review this and have a look to confirm we don't need changes into the upgrade script that would be helpful :). 

Signed-off-by: Colin Schoen <cschoen@berkeley.edu>

### CODE REVIEW

**Comment by colinschoen** [comment]
@albertlast If you have time please do review as well. I'd like another set of eyes before merging.

**Review by albertlast — APPROVED** [review]

**Comment by colinschoen** [comment]
Thanks for reviewing.

**Comment by MissAllSunday** [comment]
yep, totally forgot about displayfields. That should be the only thing to consider when upgrading

**Pull request merged**
